src/alarms/notifications/NotificationForm.js

```diff
@@ -0,0 +1,430 @@
+                "Rasters",
+                "Put an alarm on raster data"
+              ],
+              [
+                "Timeseries",
+                "Timeseries",
+                "Put an alarm on Timeseries data"
+              ]
+            ]}
+            validators={[required("Please select a data type for the alarm.")]}
+            showSearchField={false}
+            initial={
+              (
+                currentNotification &&
+                (currentNotification.raster ? "Rasters" : "Timeseries")
+              ) || "Rasters"
+            }
+          />
+          <RasterPointSelection
+            name="rasterSelection"
+            title="Raster selection"
+            validators={[rasterAndPointChosen]}
+            disabled={(formValues) => formValues.typeSelection === "Timeseries"}
+            initial={
+              (
+                currentNotification &&
+                currentNotification.raster &&
+                currentNotification.geometry && {
+                  raster: this.updateUrlToUuid(currentNotification.raster),
+                  point: {
+                    lat: currentNotification.geometry.coordinates[1],
+                    lon: currentNotification.geometry.coordinates[0]
+                  }
+                }
+              ) || null
+            }
+          />
+          <TimeseriesSelection
+            name="timeseriesSelection"
+            title="Timeseries selection"
+            subtitle="Select timeseries via asset"
+            validators={[timeseriesChosen]}
+            disabled={(formValues) => formValues.typeSelection === "Rasters"}
+            initial={
+              (
+                currentNotification &&
+                currentNotification.timeseries
+              ) || null
+            }
+          />
+          <SelectBox
+            name="relativeStartSelection"
+            title="Relative start selection"
+            subtitle="Choose relative start before or after now?"
+            choices={[
+              [
+                "Before",
+                "Before",
+                "Set the start of the simulation period from the past"
+              ],
+              [
+                "After",
+                "After",
+                "Set the start of the simulation period from now or future"
+              ]
+            ]}
+            validators={[required("Please select an option.")]}
+            showSearchField={false}
+            initial={
+              (
+                currentNotification &&
+                currentNotification.relative_start &&
+                currentNotification.relative_start.includes("-") &&
+                "Before"
+              ) || "After"
+            }
+          />
+          <RelativeField
+            name="relativeStart"
+            title="Relative start"
+            subtitle="Optional: Select the relative start of the simulation period"
+            validators={[durationValidator()]}
+            initial={
```

Comment:
I thought the Before/After select box was going to be part of the RelativeField. I think that is more logical than this situation. Then the calculation to number of seconds could be done in the field instead of here in onSubmit, and it looks more logical for the user I think.

Reply:
I think its a more logical place to have them too and it's quite a big change. I managed to do it (combine before/after select box into relative field and move all the logic in onSubmit to here as well. The most confusing part of this step to me is that I need to convert the duration value (from the API call or from the input) from a string to duration object and to number of seconds back and forth quite many times. Please have a look at them in the new commits.

Reply:
The idea should be that the field keeps its own value in whatever format is easiest for the field to use; and it should be translated to and from the APIs format when the API response is received or a new one is submitted, so once in each direction.